src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -76,6 +76,13 @@ public void start(Map<String, String> props, JestClient client) {
 
       if (client != null) {
         this.client = client;
+      } else if (config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_USERNAME)) {
+        List<String> address = config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG);
+        String username = config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_USERNAME);
+        String password = config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_PASSWORD);
+        JestClientFactory factory = new JestClientFactory();
+        factory.setHttpClientConfig(new HttpClientConfig.Builder(address).defaultCredentials(username, password).multiThreaded(true).build());
```

Comment:
This code is basically the same as the final `else` branch except you add in the `defaultCredentials(...)` call. I'd probably just combine the two branches and split the construction of the `HttpClientConfig` across multiple lines, e.g. something like:

```
JestClientFactory factory = new JestClientFactory();
List<String> addresses = config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG);
HttpClientConfig.Builder clientBuilder = new HttpClientConfig.Builder(addresses).multiThreaded(true);
String username = config.getString(ElasticsearchSinkConnectorConfig.CONNECTION_USERNAME);
if (username != null) {
  String password = config.getString(ElasticsearchSinkConnectorConfig.CONNECTION_PASSWORD);
  clientBuilder.defaultCredentials(username, password);
}
factory.setHttpClientConfig(clientBuilder.build());
this.client = factory.getObject();
```